packages/core/src/network.js

```diff
@@ -194,6 +194,21 @@ export class Network {
     // do not handle data urls
     if (request.url.startsWith('data:')) return;
 
+    // Network requests are handled by the browser, which is lenient about URL encoding.
+    // Invalid characters in the URL won't cause snapshot/build failures.
+    // Therefore, this code only checks for invalid characters in the network request URL.
+    try {
```

Comment:
I thought we were doing only for root URL and not other, here it will encode all the URLs

Reply:
- We are only modifying for root URLs, but only detects for network URLs as well.